Source/NSPasteboardItem.m

```diff
@@ -0,0 +1,177 @@
+/** <title>NSPasteboardItem</title>
+
+   <abstract>class for wrapping pasteboard content</abstract>
+
+   Copyright <copy>(C) 2017 Free Software Foundation, Inc.</copy>
+
+   Author: Daniel Ferreira <dtf@stanford.edu>
+   Date: July 2017
+
+   This file is part of the GNUstep GUI Library.
+
+   This library is free software; you can redistribute it and/or
+   modify it under the terms of the GNU Library General Public
+   License as published by the Free Software Foundation; either
+   version 2 of the License, or (at your option) any later version.
+
+   This library is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+   Library General Public License for more details.
+
+   You should have received a copy of the GNU Library General Public
+   License along with this library; see the file COPYING.LIB.
+   If not, write to the Free Software Foundation,
+   51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+*/
+
+/*
+ * FIXME: this class is useless until NSPasteboard implements
+ * NSPasteboardReading + NSPasteboardWriting.
+ */
+
+#import <AppKit/NSPasteboardItem.h>
+#import <Foundation/Foundation.h>
+
+@implementation NSPasteboardItem
+- (id)init
+{
+  if ((self = [super init]))
+    {
+      _providerMap = [[NSMutableDictionary alloc] init];
+      _dataMap = [[NSMutableDictionary alloc] init];
+      _types = [[NSMutableArray alloc] init];
+    }
+
+  return self;
+}
+
+- (NSArray *)types
+{
+  return [[NSArray arrayWithArray: _types] copy]; 
+}
+
+- (BOOL)setDataProvider:(id<NSPasteboardItemDataProvider>)dataProvider
+               forTypes:(NSArray *)types
+{
+  if (![dataProvider conformsToProtocol: @protocol(NSPasteboardItemDataProvider)])
+    {
+      NSLog(@"Pasteboard item data provider %@ must conform to"
+             "NSPasteboardItemDataProviderProtocol", dataProvider);
+      return NO;
+    }
+
+  for (NSUInteger i = 0; i < [types count]; i++)
+    {
+      NSString *type = [types objectAtIndex: i];
+      [_providerMap setObject: dataProvider forKey: type];
+    }
+
+  [_types addObjectsFromArray: types];
+  return YES;
+}
+
+- (NSString *)availableTypeFromArray:(NSArray *)types
+{
+  // FIXME: this would require a dependency on UTIs
+  return nil;
+}
+
+- (BOOL)setData:(NSData *)data forType:(NSString *)type
+{
+  if (![data isKindOfClass: [NSData class]])
+    return NO;
```

Comment:
The GNUstep coding style requires curly braces even for one line if statements to be consistent.